src/main/java/de/tum/in/www1/artemis/service/QuizExerciseService.java

```diff
@@ -347,7 +347,7 @@ public void sendQuizExerciseToSubscribedClients(QuizExercise quizExercise, Strin
             Class view = viewForStudentsInQuizExercise(quizExercise);
             byte[] payload = objectMapper.writerWithView(view).writeValueAsBytes(quizExercise);
             // For each change we send the same message. The client needs to decide how to handle the date based on the quiz status
-            if (quizExercise.isVisibleToStudents()) {
+            if (quizExercise.isVisibleToStudents() && quizExercise.hasCourse()) {
```

Comment:
Do we need this for exam quizzes? Probably not because the exercise won't be distributed by web sockets?

Reply:
I've added this to make sure that it is not called with an exam exercise. We don't really need it, but I thought it might be a good idea to prevent any errors that could happen due to setting the course to null

Reply:
I think this is ok